src/bvh/bvh_node.rs

```diff
@@ -323,6 +324,57 @@ impl<T: BHValue, const D: usize> BvhNode<T, D> {
             }
         }
     }
+
+    /// Traverses the [`Bvh`] recursively and updates the given `best_candidate` with
+    /// the nearest shape found so far.
+    ///
+    /// [`Aabb`]: ../aabb/struct.Aabb.html
+    /// [`Bvh`]: struct.Bvh.html
+    ///
+    pub fn nearest_from_recursive<'a, Shape: Bounded<T, D> + PointDistance<T, D>>(
```

Comment:
I realize that other similar functions may still be public, but AFAIK we're moving away from doing that. Let's make this new one private or `pub(crate)`, as needed.